tools/cxbe/Exe.cpp

```diff
@@ -114,6 +115,44 @@ Exe::Exe(const char *x_szFilename)
                 goto cleanup;
             }
 
+            // interpret long section names
+            if(m_SectionHeader[v].m_name[0] == '/')
+            {
+                m_SectionHeader_longname[v].m_offset = 0;
+                for(uint32 i = 1; i < 8; ++i)
+                {
+                    char c = m_SectionHeader[v].m_name[i];
+                    if(c < '0' || c > '9')
+                        break;
+                    m_SectionHeader_longname[v].m_offset *= 10;
+                    m_SectionHeader_longname[v].m_offset += c - '0';
+                }
+                m_SectionHeader_longname[v].m_longname = new char[256]();
+
+                long tmppos = ftell(ExeFile);
+                fseek(ExeFile, m_Header.m_symbol_table_addr + m_SectionHeader_longname[v].m_offset,
+                      SEEK_SET);
+
+                uint32 i = 0;
+                while(i < 255)
+                {
+                    int c = fgetc(ExeFile);
+                    if(!c || c == EOF)
+                        break;
+                    m_SectionHeader_longname[v].m_longname[i] = c;
+                    ++i;
+                }
+                m_SectionHeader_longname[v].m_longname[i] = 0;
+
+                fseek(ExeFile, tmppos, SEEK_SET);
+                printf(" (long: offset=%u name=%s) ", m_SectionHeader_longname[v].m_offset,
```

Comment:
I'm not sure how well this `printf` message fits into the output. Do you have example output?

Reply:
```
Exe::Exe: Opening Exe file...OK
Exe::Exe: Reading DOS stub...Found, Ignoring...OK
Exe::Exe: Reading PE header...OK
Exe::Exe: Reading Optional Header...OK
Exe::Exe: Reading Section Headers...
Exe::Exe: Reading Section Header 0x0000...OK 0
Exe::Exe: Reading Section Header 0x0001...OK 1
Exe::Exe: Reading Section Header 0x0002...OK 2
Exe::Exe: Reading Section Header 0x0003...OK 3
Exe::Exe: Reading Section Header 0x0004... (long: offset=4 name=$$XTIMAGE) OK 4
Exe::Exe: Reading Section Header 0x0005... (long: offset=14 name=.debug_abbrev) OK 5
Exe::Exe: Reading Section Header 0x0006... (long: offset=28 name=.debug_aranges) OK 6
Exe::Exe: Reading Section Header 0x0007... (long: offset=43 name=.debug_info) OK 7
Exe::Exe: Reading Section Header 0x0008... (long: offset=55 name=.debug_line) OK 8
Exe::Exe: Reading Section Header 0x0009... (long: offset=67 name=.debug_loc) OK 9
Exe::Exe: Reading Section Header 0x000A... (long: offset=78 name=.debug_ranges) OK 10
Exe::Exe: Reading Section Header 0x000B... (long: offset=92 name=.debug_str) OK 11
Exe::Exe: Reading Sections...
Exe::Exe: Reading Section 0x0000...OK
Exe::Exe: Reading Section 0x0001...OK
Exe::Exe: Reading Section 0x0002...OK
...
```

Reply:
I could move it to after the OK

Reply:
I don't think the message adds anything (but can quickly become an issue if someone sneaks bad symbols into section names, such as `\n` or `\r`, VT markup etc.).

There's also a inconsistency that section names are not shown for short sections.

So I'd remove the message for now.